Code/GraphMol/SubstructLibrary/SubstructLibrary.h

```diff
@@ -678,6 +741,19 @@ class RDKIT_SUBSTRUCTLIBRARY_EXPORT SubstructLibrary {
     return rdcast<unsigned int>(molholder->size());
   }
 
+  //! does error checking
+  void setSearchOrder(const std::vector<unsigned int> &order) {
+    for (const auto idx : order) {
```

Comment:
We should probably also check that order.size() <= mols.size() and != 0 as well as a PRECONDITION.

Reply:
I don't think either of those conditions would cause the search to misbehave, so they would at best be warnings.